### PR TITLE
chore(Table): Add expanded state to `Table -> ControlledState` story

### PR DESCRIPTION
**Pull request by rohan-kadkol**
Like @bentleyvk suggested, added expanded state controls in the `Table -> ControlledState` story according to a related [CodeSandbox example](https://codesandbox.io/s/table-control-select-and-expand-state-oju3lx).

## Checklist

- ~[ ]~ Add meaningful unit tests for your component (verify that all lines are covered)
- [x] Verify that all existing tests pass
- ~[ ]~ Add component features demo in Storybook (different stories)
- ~[ ]~ Approve test images for new stories
- ~[ ]~ Add screenshots of the key elements of the component

### CODE REVIEW

File: apps/storybook/src/Table.stories.tsx [hunk 1]
```diff
@@ -1344,20 +1402,42 @@ export const ControlledState: Story<Partial<TableProps>> = (args) => {
...
+      </InputGroup>
       <Table
         columns={columns}
-        data={data}
         emptyTableContent='No data.'
         useControlledState={controlledState}
         stateReducer={tableStateReducer}
         isSelectable
         {...args}
+        data={data}
```
Comment (rohan-kadkol) [inline apps/storybook/src/Table.stories.tsx:1432]: This change is to prevent the default `args.data` from overriding the custom `data`

**Comment by mayank99** [comment]
Did someone ask for this? I dont see any issue.

**Comment by rohan-kadkol** [comment]
> Did someone ask for this? I dont see any issue.

Yes, @bentleyvk suggested changing the controlled state story to something similar to the [CodeSandbox example](https://codesandbox.io/s/table-control-select-and-expand-state-oju3lx).

**Review by mayank99 — APPROVED** [review]

**Comment by bentleyvk** [comment]
You got me wrong 😀 What I meant was to not to use `useControlledState` prop since it requires users to fully control table state (which has a lot of nuances) and change the story with something using table instance that is being get from `stateReducer` like in the codesandbox that you linked in description.

**Pull request merged**